astroid/rebuilder.py

```diff
@@ -140,14 +145,15 @@ def visit_module(
 
         Note: Method not called by 'visit'
         """
-        node, doc = self._get_doc(node)
+        doc, doc_node = self._get_doc(node)
```

Comment:
Here `doc_node` will still be an ast node. Since we need to create the parent, i.e. `Module` node first, we can't pass it as argument to `Module.__init__`. Instead move `doc_node` to `postinit`.
The argument should be something like this
```py
newnode.postinit(
    ...,
    doc_node=self.visit(doc_node, newnode),
)